src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -465,7 +466,9 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
 
                         // UNDONE: (Refactor) Refactor TargetUpToDateChecker to take a logging context, not a logging service.
                         TargetUpToDateChecker dependencyAnalyzer = new TargetUpToDateChecker(requestEntry.RequestConfiguration.Project, _target, targetLoggingContext.LoggingService, targetLoggingContext.BuildEventContext);
+                        MSBuildEventSource.Log.PerformDependencyAnalysisStart();
```

Comment:
I'd claim that creating the object is part of the time spent (I doubt this matters much though).

```suggestion
                        MSBuildEventSource.Log.PerformDependencyAnalysisStart();
                        TargetUpToDateChecker dependencyAnalyzer = new TargetUpToDateChecker(requestEntry.RequestConfiguration.Project, _target, targetLoggingContext.LoggingService, targetLoggingContext.BuildEventContext);
```